gulpfile.js

```diff
@@ -129,26 +146,35 @@ gulp.task('build-ci', ['clean'], function (cb) {
 });
 
 gulp.task('build-test', function buildTest(cb) {
-    var config = _.defaultsDeep({}, require('./webpack.config'));
-
-    config.entry = {};
-    // the key in the entry, formed by path.join, determines the target folder structure of the test file
-    config.entry[path.join('request', 'repository', 'request-repository-cache')] = './test/request/repository/request-repository-cache.spec.ts';
-
-    config.output.path = testSettings.output;
-
-    var started = false;
-    function processResult(err, stats) {
-        gutil.log('Webpack\n' + stats.toString(config.log));
-
-        if (!started) {
-            started = true;
-            cb();
+    getTests(function collectAllTests(tests) {
+        var config = _.defaultsDeep({}, require('./webpack.config'));
+        var pathPrefix = path.join(testSettings.input, '/');
+        var re = new RegExp(testSettings.testSuffix + '$');
```

Comment:
Converted testSuffix to regex. 

Kept the logic in the same place since `getTests` returns the array of full path names and here we simply strip out the suffix.